frame/example-offchain-worker/src/lib.rs

```diff
@@ -149,6 +161,22 @@ decl_storage! {
 }
 
 impl<T: Trait> Module<T> {
+	fn check_if_should_proceed(_block_number: T::BlockNumber) -> Result<(), &'static str> {
+		let s_next_at = StorageValueRef::persistent(b"example-offchain-worker::next-at"); // TODO: Rename after OCW renamed
+
+		if let Some(Some(next_at)) = s_next_at.get::<T::BlockNumber>() {
+			if next_at > _block_number {
+				return Err("[OCW] Too early to execute OCW. Skipping");
+			}
+		}
+
+		let current_block = <frame_system::Module<T>>::block_number();
```

Comment:
The block number is given as an argument to offchain_worker(block_number), it might be safer to use this parameter instead.